test/utils/deployment.ts

```diff
@@ -0,0 +1,172 @@
+        yarnStartProcess.stderr.pipe(errorStream);
+    }
+    // Wait for the API to boot up
+    await waitForApiStartupAsync(yarnStartProcess);
+}
+
+/**
+ * Tears down the old 0x-api instance.
+ */
+export async function teardownApiAsync(): Promise<void> {
+    if (!yarnStartProcess) {
+        throw new Error('There is no 0x-api instance to tear down');
+    }
+    yarnStartProcess.kill();
+    await teardownDependenciesAsync();
+}
+
+/**
+ * Sets up 0x-api's dependencies.
+ * @param logType Indicates where logs should be directed.
+ */
+export async function setupDependenciesAsync(logType: LogType = LogType.Hidden): Promise<void> {
+    // Remove the saved volumes to ensure a sandboxed testing environment.
+    await rimrafAsync(`${apiRootDir}/0x_mesh`);
+    await rimrafAsync(`${apiRootDir}/postgres`);
+
+    // Spin up the 0x-api dependencies
+    const up = spawn('docker-compose', ['up'], {
+        cwd: apiRootDir,
+        env: {
+            ...process.env,
+            ETHEREUM_RPC_URL: 'http://ganache:8545',
+            ETHEREUM_CHAIN_ID: '1337',
+        },
+    });
+
+    // Direct the logs to the appropriate locations.
+    if (logType === LogType.Console) {
+        up.stdout.on('data', chunk => {
+            neatlyPrintChunk('[docker-compose up]', chunk);
+        });
+        up.stderr.on('data', chunk => {
+            neatlyPrintChunk('[docker-compose up | error]', chunk);
+        });
+    } else if (logType === LogType.File) {
+        const logStream = fs.createWriteStream(`${apiRootDir}/dependency_logs`, { flags: 'a' });
+        const errorStream = fs.createWriteStream(`${apiRootDir}/dependency_errors`, { flags: 'a' });
+        up.stdout.pipe(logStream);
+        up.stderr.pipe(errorStream);
+    }
+
+    // Wait for the dependencies to boot up.
+    await waitForDependencyStartupAsync(up);
+}
+
+/**
+ * Tears down 0x-api's dependencies.
+ */
+export async function teardownDependenciesAsync(): Promise<void> {
+    const down = spawn('docker-compose', ['down'], {
+        cwd: apiRootDir,
+    });
+    await new Promise<void>(resolve => {
+        down.on('close', () => {
+            resolve();
+        });
+    });
+    await rimrafAsync(`${apiRootDir}/0x_mesh`);
+    await rimrafAsync(`${apiRootDir}/postgres`);
+}
+
+function neatlyPrintChunk(prefix: string, chunk: Buffer): void {
+    const data = chunk.toString().split('\n');
+    data.filter((datum: string) => datum !== '').map((datum: string) => {
+        log.log(prefix, datum.trim());
+    });
+}
+
+async function waitForApiStartupAsync(logStream: ChildProcessWithoutNullStreams): Promise<void> {
+    return new Promise<void>(resolve => {
+        logStream.stdout.on('data', (chunk: Buffer) => {
+            const data = chunk.toString().split('\n');
+            for (const datum of data) {
+                if (/API (HTTP) listening on port 3000!/.test(datum)) {
+                    resolve();
+                }
+            }
+        });
+    });
```

Comment:
This could be another accidental awaiting forever case. Should we add a timeout?